### PR TITLE
Adds a way to retrieve all entries matching a lookup of idb tables

### PR DESCRIPTION
**Pull request by sabieber**
Previously there were only ways to easily either get all entries of a table or lookup a single entry.

Fixes: OX-6913

### CODE REVIEW

File: src/main/java/sirius/biz/codelists/CodeListLookupTable.java [hunk 1]
```diff
@@ -131,4 +131,9 @@ private boolean filter(CodeListEntry<?, ?> entry, String searchTerm, String lang
     protected Stream<LookupTableEntry> performScan(String lang) {
         return codeLists.getEntries(codeList).stream().map(entry -> extractEntryData(entry, lang));
     }
+
+    @Override
+    protected Stream<LookupTableEntry> performLookupScan(String lang, String lookupPath, String lookupValue) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/codelists/CodeListLookupTable.java:136]: Optional: "performQuery" - scan is used to indicate that "no filter is applied at all"

File: src/main/java/sirius/biz/codelists/LookupTable.java [hunk 2]
```diff
@@ -731,4 +730,22 @@ public boolean canScan() {
...
+    /**
+     * Enumerates all entries matching the given lookup in the table using the given language.
+     *
+     * @param lang        the language to translate the name and description to
+     * @param lookupPath  the field to search in
+     * @param lookupValue the value to search for
+     * @return a stream of all entries matching the lookup or an empty stream if scanning isn't supported
+     */
+    public Stream<LookupTableEntry> lookupScan(String lang, String lookupPath, String lookupValue) {
+        if (!canScan()) {
```
Comment (andyHa) [inline src/main/java/sirius/biz/codelists/LookupTable.java:743]: IMHO this check can be skipped. "canScan" is rather a "is it smart to scan" as some lists are very big and should not be output completeley. However, as a filter is applied here, we can (hopefully) safely always perform a query

**Review by idlira — APPROVED** [review]

File: src/main/java/sirius/biz/codelists/CodeListLookupTable.java [hunk 3]
```diff
@@ -131,4 +131,9 @@ private boolean filter(CodeListEntry<?, ?> entry, String searchTerm, String lang
     protected Stream<LookupTableEntry> performScan(String lang) {
         return codeLists.getEntries(codeList).stream().map(entry -> extractEntryData(entry, lang));
     }
+
+    @Override
+    protected Stream<LookupTableEntry> performQuery(String lang, String lookupPath, String lookupValue) {
+        throw new UnsupportedOperationException();
```
Comment (meg1502) [inline src/main/java/sirius/biz/codelists/CodeListLookupTable.java:137]: Maybe a a small comment why this is unsupported

mainly to differiantiate between "not yet implemented" and "this method will never work on this class"

**Review by meg1502 — APPROVED** [review]

File: src/main/java/sirius/biz/codelists/IDBLookupTable.java [hunk 4]
```diff
@@ -48,6 +48,7 @@
     private static final String CACHE_PREFIX_NORMALIZE_WITH_MAPPING = "normalize-with-mapping";
     private static final String CACHE_PREFIX_REVERSE_LOOKUP = "reverse-lookup-";
     private static final String CACHE_PREFIX_FETCH_OBJECT = "fetch-object-";
+    private static final String CACHE_PREFIX_LOOKUP_SCAN = "lookup-scan-";
```
Comment (andyHa) [inline src/main/java/sirius/biz/codelists/IDBLookupTable.java:51]: not used (?)

**Review by andyHa — APPROVED** [review]

**Review by cschierle — APPROVED** [review]

**Review by ymo-sci — APPROVED** [review]

**Review by qw3ry — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**